### ISSUE TITLE
Training does not start

### ISSUE THREAD

**Issue by LoekvanKooten** [modernmt/modernmt#513]
`./mmt create zh nl data/procdut -e procdut --resume --tensorboard-port 8066
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:526: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:527: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:528: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:529: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:530: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:535: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])

=========== TRAINING STARTED ===========

ENGINE:   procdut
LANGUAGE: zh > nl

(1/4) Cleaning corpora
(1/4) Cleaning corpora SKIPPED
(2/4) Creating engine

ERROR Unexpected exception: [Errno 2] No such file or directory: '/home/loek/modernmt2/engines/procdut/engine.xconf'
`

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

as I understood, you are resuming a previous training. right?

If no, i.e. if you want to start a training from scratch, you should remove anything related to `procdut ` both in `engines` and in `runtime`.

If yes, I would like to know a few things in order to debug:
- at which step does it stop?
- which version of ModernMT are you using?


Meanwhile, here is a trick to let you overcome this problem by copying into `engines/procdut` the `engine.xconf` created for another engine, paying attention to set the language correctly.

**Comment by LoekvanKooten** [comment]
I trained first for an hour (it's a very small experimental database), but that went haywire in step 1/2 of the neural training. Something with not being able to find tensorfile files (I lost the exact error, I'm afraid). I ended up with an engine that could be started, but would show STOPPED in status nonetheless. 

So this is a completely new attempt. This also happens when engines/procdut and runtime/procdut are completely empty.

I used the engine.xconf trick before, but assumed that editing exactly this file may have caused the stop mentioned above in the first place.

**Comment by LoekvanKooten** [comment]
@nicolabertoldi  

P.S. This is 4.2

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 
so your are saying that you want to train an engine from scratch.

In this case you do not need to specify the parameter "--resume".

**Comment by LoekvanKooten** [comment]
(Or was that --debug?)

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

I try to clarify:
- when you want to start a training from scratch, you must not use the parameter `--resume`
- when you want to resume a interrupted training, you have to use `--resume`, so that the system does not re-compute the training steps already done. Note that:
-- the interruption  may happen after the training step 2 (`(2/4) Creating engine`), which actually generates the file `engine.xconf`
-- temporary or final files must not be removed

With the parameter `--debug`, the system does not delete the runtime directory after a successful training.

**Comment by LoekvanKooten** [comment]
Ah, okay, so any training can be resumed later if it has been interrupted for whatever reason, correct?

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

Not exactly,

A training is considered successful in two cases:
- if the termination policy is reached autonomously
- if the training is interrupted by the user after the generation of at least one checkpoint.

Said that, the resuming works as follows:
- a failed training can be resumed always, 
- a successful training performed without the parameter `--debug` cannot be resumed
- a successful training performed with the parameter `--debug` can be resumed always

A training is successful in two cases:
- if the termination policy is reached autonomously
- if the training is interrupted by the user after the generation of at least one checkpoint.

Obviously, if a training succeeded because the termination policy is fired, its resuming will succeeds immediately if the termination policy does not change.

**Comment by LoekvanKooten** [comment]
So if the training is interrupted by an error, it cannot be resumed, right?

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

No,
if the training is interrupted by an error (i.e. if it is failed), it **can** be resumed

**Comment by LoekvanKooten** [comment]
Ah, "a failed training can be resumed always" (no matter which flag you use, right).

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

correct.

the important thing is that, in case of failure, files must not be deleted.

**Comment by LoekvanKooten** [comment]
It happened again.

@nicolabertoldi 

` ./mmt create zh nl data/procdut -e procdut --debug --tensorboard-port 8066
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:526: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:527: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:528: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:529: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:530: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/home/loek/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:535: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])

=========== TRAINING STARTED ===========

ENGINE:   procdut
LANGUAGE: zh > nl

(1/4) Cleaning corpora
    (1/6) Cleaning corpora...                                    DONE in 0s
    (2/6) De-duplicate corpora...                                DONE in 0s
    (3/6) Preprocess corpora...                                  DONE in 0s
    (4/6) Creating aligner filter...                             DONE in 0s
    (5/6) Scoring corpora...                                     DONE in 0s
    (6/6) Apply aligner-based filter...                          DONE in 0s
(1/4) Cleaning corpora DONE in 2s
(2/4) Creating engine
(2/4) Creating engine DONE in 0s
(3/4) Generating binary archives
    (1/4) Tokenize corpora...                                    DONE in 0s
    (2/4) Training BPE model...                                  DONE in 3s
    (3/4) Encoding training corpora...                           DONE in 0s
    (4/4) Generating binary data...                              DONE in 0s
(3/4) Generating binary archives DONE in 5s
(4/4) Training neural model
    (1/2) Train neural network...                                
ERROR Unexpected exception: Command 'fairseq-train /home/loek/modernmt2/runtime/procdut/tmp/training/data_generated --save-dir /home/loek/modernmt2/runtime/procdut/tmp/training/_temp_train/nn_model --task mmt_translation --user-dir /home/loek/modernmt2/build/mmt-4.2.jar/mmt --share-all-embeddings --no-progress-bar --tensorboard-logdir /home/loek/modernmt2/runtime/procdut/tmp/training/_temp_train/tensorboard_logdir --dataset-impl mmap --arch transformer_mmt_base --clip-norm 0.0 --label-smoothing 0.1 --attention-dropout 0.1 --dropout 0.3 --weight-decay 0.0 --criterion label_smoothed_cross_entropy --optimizer adam --adam-betas (0.9, 0.98) --log-interval 100 --lr 0.0005 --lr-scheduler inverse_sqrt --min-lr 1e-09 --warmup-init-lr 1e-07 --warmup-updates 4000 --max-tokens 3072 --update-freq 4 --save-interval-updates 1000 --keep-interval-updates 10 --keep-last-epochs 10' failed with exit code 1
`

**Comment by nicolabertoldi** [comment]
@LoekvanKooten 

I suppose this is the original problem which interrupted your training.

and it is clearly non related to the subject of this issue.

Could you please close this issue,
and open a new one reporting the error.

I will try to answer there.

thank you!

**Comment by LoekvanKooten** [comment]
Will do right away.
